### DISCUSSION TITLE
Change password

### DISCUSSION THREAD

**Discussion by worksofliam** [codefori/vscode-ibmi#932] [Ideas]
Would it be cool to change your password after connecting to a system?

https://www.ibm.com/docs/en/i/7.3?topic=ssw_ibm_i_73/apis/QSYCHGPW.html

**Comment by worksofliam** [comment]
Would it be cool to prompt the user how many days they have left before their password expires? Is there an API for that?

**Comment by Wright4i** [comment]
It would be cool! And yes there's an easy way to grab # of days:

```sql
select 
  days_until_password_expires 
from 
  qsys2.user_info 
where 
  authorization_name = current_user
```

**Reply by Wright4i** [reply]
null password never expires. 0 password is expired. >0 # of days

**Comment by nick-roux** [comment]
The following will be very useful:

- On connecting to a system, check password expiry day and display Popup Notification if password expires within the next 30 days (maybe make the number of days configurable)
- Possibility of clicking on the popup to change password

We have users configured for Single SignOn, so they never see password expiry notifications on 5250 sessions on iACS.
Since Code for IBM i uses ssh and reply on the local password on the LPAR, developers occasionally disable their profiles because the password has expired without them being warned about it.

**Comment by sebjulliand** [comment]
It's been addressed by #3055 .

**Reply by nick-roux** [reply]
This looks brilliant, thank you
